R/manage_additional_metadata.R

```diff
@@ -0,0 +1,193 @@
+  x[c(drug_name2, conc2, drug2)] <- NULL
+  # remove special class
+  class(x) <- setdiff(class(x), "drug_name2")
+  return(x)
+}
+
+#' @export
+#' @describeIn modifyData includes the data source
+modifyData.data_source <- function(x, option, keep, ...) {
+  assert_data_table(x)
+  assert_string(option)
+  assert_choice(option, c("average", "toDrug", "toCellLine"))
+  assert_string(keep, null.ok = TRUE)
+  
+  pidfs <- get_prettified_identifiers(simplify = TRUE)
+  dt_src <- pidfs[["data_source"]]
+  drug <- pidfs[["drug_name"]]
+  clid <- pidfs[["cellline"]]
+  cl_name <- pidfs[["cellline_name"]]
+  
+  if (option == "average") {
+    # drop data and keep only the requested value
+    x <- average_biological_replicates_dt(x, dt_src)
+  } else {
+    duplicated_rows <- get_duplicated_rows(x, c(drug, clid))
+    if (option == "toDrug") {
+      drugs_to_combine <- unique(x[duplicated_rows, drug])
+      drug_idx <- which(x[[drug]] %in% drugs_to_combine)
+      drug_to_replace <- x[drug_idx, drug]
+      x[drug_idx, drug] <-
+        vapply(
+          seq_len(length(drug_to_replace)),
+          function(y) sprintf("%s (%s)", drug_to_replace[y], x[, dt_src][y]), "string")
+    } else if (option == "toCellLine") {
+      cell_lines_to_combine <- unique(x[duplicated_rows, cl_name])
+      cell_line_idx <- which(x[[cl_name]] %in% cell_lines_to_combine)
+      x[cell_line_idx, cl_name] <-
+        sprintf("%s (%s)", x[cell_line_idx, cl_name], x[cell_line_idx, dt_src])
+    }
+  }
+  # drop the additional columns
+  x[, c(dt_src) := NULL]
+  # remove special class
+  class(x) <- setdiff(class(x), dt_src)
+  return(x)
+}
+
+
+#' @export
+#' @describeIn modifyData includes the name of other additional variables
+#' #
+modifyData.default <- function(x, option, keep, ...) {
+  assert_data_table(x)
+  assert_string(option)
+  assert_choice(option, c("average", "toDrug", "toCellLine"))
+  assert_string(keep, null.ok = TRUE)
+  pidfs <- get_prettified_identifiers(simplify = TRUE)
+  additional_var_names <- class(x)[[1]]
+  additional_var <- ifelse(additional_var_names %in% names(pidfs),
+                           pidfs[[additional_var_names]],
+                           additional_var_names)
+  cell_name <- pidfs[["cellline_name"]]
+  drug_name <- pidfs[["drug_name"]]
+  
+  if (option == "average") {
+    # drop data and keep only the requested value
+    x <- average_biological_replicates_dt(x, additional_var)
+  } else {
+    if (option == "toDrug") {
+      x <- modify_label(x, drug_name, additional_var)
+    } else if (option == "toCellLine") {
+      x <- modify_label(x, cell_name, additional_var)
+    }
+  }
+  # drop the additional columns
+  if ("data.table" %in% class(x)) {
+    x[[additional_var]] <- NULL
+  } else {
+    x[additional_var] <- NULL
+  }
+  # remove special class
+  class(x) <- setdiff(class(x), additional_var_names)
+  return(x)
+}
+
+#' @keywords internal
+modify_label <- function(x, option, var_name) {
```

Comment:
This is an internal function, but maybe you could add tests to show how this function is supposed to work?